### PR TITLE
add command goc list

### PR DESCRIPTION
**Pull request by tongjingran**

### CODE REVIEW

File: pkg/client/client_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/sirupsen/logrus"
```
Comment (lyyyuna) [inline pkg/client/client_test.go:11]: 没有必要引入其它的日志库 logrus。

这里打日志也没啥必要吧

File: pkg/client/client.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+func (c *client) ListAgents(wide bool) {
+	u := fmt.Sprintf("%s%s", c.Host, CoverAgentsListAPI)
+	_, body, err := c.do("GET", u, "", nil)
+	if err != nil && isNetworkError(err) {
+		_, body, err = c.do("GET", u, "", nil)
+	}
+	if err != nil {
+		err = fmt.Errorf("%w: %v", ERROR_GOC_LIST, err)
+		log.Fatalf(err.Error())
```
Comment (lyyyuna) [inline pkg/client/client.go:73]: fatalf("go list failed: %v", err)

不需要再 err 封装一层

File: pkg/client/client.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	Hostname string `json:"hostname"`
+	CmdLine  string `json:"cmdline"`
+	Pid      string `json:"pid"`
+}
+
+// NewWorker creates a worker to contact with host
+func NewWorker(host string) Action {
+	_, err := url.ParseRequestURI(host)
+	if err != nil {
+		log.Fatalf("Parse url %s failed, err: %v", host, err)
```
Comment (lyyyuna) [inline pkg/client/client.go:57]: 日志我们统一下风格，小写开头。

File: pkg/client/client_test.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	assert.Contains(t, err.Error(), "invalid method")
+}
+
+func TestGetSimpleSvcName(t *testing.T) {
+	testCases := map[string]struct {
+		input    string
+		expected string
+	}{
+		"only file":       {"file.go", "file.go"},
+		"only path":       {"/local/path/file", "file"},
```
Comment (lyyyuna) [inline pkg/client/client_test.go:90]: 参考 https://stackoverflow.com/questions/28387378/go-test-file-for-specific-os-or-architecture

给 Windows 增加特殊的测试用例。

File: pkg/client/client_test.go (hunk 1 above)
Comment (tongjingran) [reply pkg/client/client_test.go:11]: 是为了单测抓输出到控制台的信息，复制自https://github.com/qiniu/goc/blob/7da52236182f82e0c8a5279d19071a9661309963/cmd/diff_test.go#L34

Comment (lyyyuna) [reply pkg/client/client_test.go:11]: 不需要通过日志库来打。去掉 logrus

**Comment by tongjingran** [comment]
use ` goc list ` command, we will get response like this:
`
  __ _  ___   ___ 
 / _  |/ _ \ / __|
| (_| | (_) | (__ 
 \__, |\___/ \___|
 |___/ 

ID             REMOTEIP    CMD                                                     
e3a1c4e45b8c   127.0.0.1   ./simple-go-server/simple-go-server --listen-addr :5002   
c9d1c5dca5df   127.0.0.1   ./simple-go-server/simple-go-server --listen-addr :5003  
`

use ` goc list --wide ` command, we will get response like this:
`

  __ _  ___   ___ 
 / _  |/ _ \ / __|
| (_| | (_) | (__ 
 \__, |\___/ \___|
 |___/ 

ID             REMOTEIP    HOSTNAME            PID    CMD                                                     
e3a1c4e45b8c   127.0.0.1   MacBook-Pro.local   5851   ./simple-go-server/simple-go-server --listen-addr :5002   
c9d1c5dca5df   127.0.0.1   MacBook-Pro.local   6194   ./simple-go-server/simple-go-server --listen-addr :5003  
`

**Review by lyyyuna — APPROVED** [review]

**Pull request merged**